accepted/2021/net6.0-tfms/net6.0-tfms.md

```diff
@@ -0,0 +1,159 @@
+|                    | (+everything else inherited from net6.0) |
+
+### Compatibility rules
+
+The [built-in compatibility rules][net5.0] of the `netX.Y-platform` TFMs make a
+TFM automatically compatible with:
+
+1. Itself
+2. Earlier versions of itself
+3. The corresponding `netX.Y` TFM
+4. Earlier versions of the corresponding `netX.Y` TFM
+
+The `net5.0` TFM has the additional twist that it's compatible with
+`netcoreapp3.2` (and earlier), `netstandard2.1` (and earlier) as well as .NET
+Framework (`net4x` and earlier, but under a warning). The .NET Framework
+compatibility is achieved as a *fallback*, that is, it's only used when nothing
+else applies.
+
+We don't want to model the Xamarin TFMs as a fallback but instead want them to
+apply **before** `netcoreapp` and `netstandard`. Usually the existence of the
+Xamarin TFM implementation is to bait a `netstandard` API and switch the
+platform specific implementation. Most of the Plugin type packages use this
+pattern and in fact the `netstandard` "implementation" is more like a reference
+assembly with just stubs that throw at runtime. `Xamarin.Essentials` is a good
+example of this. The same rational can be applied to `netcoreapp` where this
+would logically be the server implementation (or maybe a .NET Core 3.x desktop
+implementation).
+
+A `net6.0-maccatalyst` project referencing a NuGet package should behave as
+follows:
+
+* It should prefer `xamarin.ios` assets over `netcoreapp` and `netstandard`
+  assets.
+    - However, it should still prefer `net5.0`/`net6.0` assets over
+      `xamarin.ios`.
+    - It also shouldn't be compatible with any other existing Xamarin TFM (such
+      as `xamarin.mac`)
+* Generate NuGet warning [NU1701] when a `xamarin.ios` asset is being used
+    - Package 'packageId' was restored using 'xamarin.ios' instead the project
+      target framework 'net6.0-maccatalyst'. This package may not be fully
+      compatible with your project.
+* Should only use compatible `net5.0` based TFMs, namely `net5.0`, `net6.0`, and
+  `net6.0-maccatalyst`.
+    - Specifically, it should not accept `net6.0-ios`. The expectation is that
+      moving forward libraries that want to work on iOS and Mac Catalyst should
+      use `net6.0` or multi-target for `net6.0-ios` and `net6.0-maccatalyst`
+
+## APIs
+
+We need to add platform detection APIs for Mac Catalyst:
+
+```C#
+namespace System
+{
+    public partial class OperatingSystem
+    {
+        public static bool IsMacCatalyst();
+        public static bool IsMacCatalystVersionAtLeast(int major, int minor = 0, int build = 0);
+    }
+}
+```
+
+## RIDs
+
+We need to add RIDs for all these platforms:
+
+* `net6.0-android`
+    - TBD
+* `net6.0-ios`
+    - TBD
+* `net6.0-maccatalyst`
+    - `maccatalyst-arm64`
+    - `maccatalyst-arm64-13`
+    - `maccatalyst-arm64-14`
+    - `maccatalyst-x64`
+    - `maccatalyst-x64-13`
+    - `maccatalyst-x64-14`
+* `net6.0-macos`
+    - TBD
+* `net6.0-tvos`
+    - TBD
```

Comment:
@marek-safar What are the RIDs we need for Android, iOS, macOS, and tvOS?

Reply:
There is a lot of them. If you want to list them all you can start with listing everything in

Android-arch: x64;x86;arm;arm64
Android-versions: 21 to 29 range
ios/tv-arch: arm64;x64
ios/tv-versions: 11 to 14 range
ios-arch: arm
ios-vesion: 10 only